### PR TITLE
Ensure that resteasy rest-client uses QuarkusInjectorFactory as well

### LINKED ISSUE #2773: NullPointerException using @Provider in MP Rest Client
**Describe the bug**
@Inject doesn't work in @Provider when the Provider is used by MP Rest Client

**Expected behavior**
The Bean injected

**Actual behavior**
The Bean field is null

**To Reproduce**
1. git clone https://github.com/masini/quarkus-rest-client-npe
2. mvn package
3. Server Test Works, Client Test Fails

### PR DESCRIPTION
**Pull request by geoand**
This is done in order to be able to use `@Provider` annotated `ContextResolver` classes as beans - same as is the case for RESTEasy server extensions.

Fixes #2773

### CODE REVIEW

**Comment by geoand** [comment]
The only thing I am a little concerned about is setting `QuarkusInjectorFactory.CONTAINER` twice (if the rest client extension is in play). I don't think it will ever be a problem but if you think we should add some synchronization, then I can of course do that

File: extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java [hunk 1]
```diff
@@ -24,8 +30,12 @@ public void setSslEnabled(boolean sslEnabled) {
         RestClientBuilderImpl.setSslEnabled(sslEnabled);
     }
 
-    public void initializeResteasyProviderFactory(boolean useBuiltIn, Set<String> providersToRegister,
+    public void initializeResteasyProviderFactory(BeanContainer beanContainer,
+            List<Function<Object, Object>> propertyUnwrappers, boolean useBuiltIn,
+            Set<String> providersToRegister,
             Set<String> contributedProviders) {
+        QuarkusInjectorFactory.CONTAINER = beanContainer;
```
Comment (mkouba) [inline extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java]: We should probably move this (and the code from `ResteasyServerCommonRecorder#setupIntegration()`) to the `ResteasyCommonProcessor`. Actually, we should probably move the `ResteasyServerCommonProcessor#setupInjection()` method and this build step should depend on `ResteasyInjectionReadyBuildItem`, or?

Comment (geoand) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java]: Interesting idea, I like it. @gsmet do you agree?

Comment (mkouba) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java]: In fact, `ResteasyInjectionReadyBuildItem` could hold a `RuntimeValue<org.jboss.resteasy.spi.InjectorFactory>` reference and we could keep the `QuarkusInjectorFactory` fields package-private.

Comment (geoand) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java]: Should be fixed now

**Review by gsmet — CHANGES_REQUESTED** [review]
It looks good to me, nice work.

I just had a comment about a comment where you totally changed the meaning of the initial wording, and the new meaning is IMHO incorrect. Can you check?

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyInjectionReadyBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package io.quarkus.resteasy.common.deployment;
+
+import org.jboss.resteasy.spi.InjectorFactory;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+
+/**
+ * Gives access to the configured {@link InjectorFactory}
+ * Can also be used as a marker that RESTEasy has been properly setup and is ready for use
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyInjectionReadyBuildItem.java]: Of course yes

[the remark this answers is not in the document]
Comment (geoand) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyInjectionReadyBuildItem.java]: Fixed

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
It's OK for me. @mkouba I let you review the latest changes and approve too?

**Review by mkouba — APPROVED** [review]
@geoand That's exactly what I meant. Good job!

**Comment by geoand** [comment]
Thanks for your help @mkouba and @gsmet !

**Pull request merged**